### PR TITLE
BAH-977: Modified Profile Resource to support relationships both ways

### PR DESCRIPTION
**Pull request by Andu033**

### CODE REVIEW

File: bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java [hunk 1]
```diff
@@ -246,13 +246,23 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
...
-
+    
     private Relationship createRelationship(Person currentPerson, Map<String, Object> relationshipJson) {
-        Relationship relationship = new Relationship(currentPerson,
-                getPerson((Map<String, Object>) relationshipJson.get("personB")),
-                getRelationshipType((Map<String, Object>) relationshipJson.get("relationshipType")));
+        Relationship relationship;
+        String direction = (String) relationshipJson.get("direction");
+        
+        if (direction.equals("aToB")) {
```
Comment (rbuisson) [inline bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java]: I would rather use a `Constant` or better, an `Enum` to store those keywords.
And rename to `A_to_B` and `B_to_A`.

File: bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java [hunk 2]
```diff
@@ -246,13 +246,23 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
...
+            relationship = new Relationship(currentPerson,
+                    getPerson((Map<String, Object>) relationshipJson.get("person")),
```
Comment (rbuisson) [inline bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java]: `getPerson()` and `getRelationshipType()` can be written only once, before the `if`.

Comment (rbuisson) [reply bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java]: Could you implement a line where we make it backward compatible?
Something that would check if `relationshipJson.get("person"))` is null, and if so try to get `relationshipJson.get("personB"))`. If the latter is not null, we set the direction to be `A_to_B`.

File: bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java [hunk 3]
```diff
@@ -246,13 +246,23 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
...
+                    getRelationshipType((Map<String, Object>) relationshipJson.get("relationshipType")));
+        } else {
```
Comment (rbuisson) [inline bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java]: In general, I would do the opposite. The default should assume that we are in a `A_to_B` situation, not the other way.

File: bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java [hunk 4]
```diff
@@ -246,15 +246,37 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
...
         relationship.setEndDate(new DateMapper().convertUTCToDate(getValueFromMap(relationshipJson, "endDate")));
-
+        
         return relationship;
     }
+    
+    enum RelationshipDirection {
+        A_TO_B,
+        B_TO_A
+    }
```
Comment (rbuisson) [inline bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java]: @Andu033 , could you add that at the very top of the Controller instead? (and make it private)

File: bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java (hunk 3 above)
Comment (rbuisson) [reply bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java]: @Andu033 , this is still not in the right way. I would do the opposite of doing something specific only when the case is specificall `B_TO_A`. All other cases would be considered `A_TO_B`. Does that make sense?

**Comment by Andu033** [comment]
@rbuisson do you mean to add test that verify the changes? Because i don't see test that test the previous behaviour of the creating proccess.

**Comment by rbuisson** [comment]
Work on this has been interrupted. Closing the PR.

**Pull request closed without merging**